tests/test_match.py

```diff
@@ -749,3 +788,92 @@ def test_match_os_any():
         0x0,
     )
     assert "test rule" in matches
+
+
+def test_index_features_and_unstable():
+    rule = textwrap.dedent(
+        """
+        rule:
+            meta:
+                name: test rule
+                scopes:
+                    static: function
+                    dynamic: process
+            features:
+                - and:
+                    - mnemonic: mov
+                    - api: CreateFileW
+        """
+    )
+    r = capa.rules.Rule.from_yaml(rule)
+    rr = capa.rules.RuleSet([r])
+    index: capa.rules.RuleSet._RuleFeatureIndex = rr._feature_indexes_by_scopes[capa.rules.Scope.FUNCTION]
+
+    # there's a single rule, and its indexed by a single feature
```

Comment:
These comments and below are really helpful in understanding the intention with indexing features, thank you